### DISCUSSION TITLE
Add an explicit method to remove the objective

### DISCUSSION THREAD

**Discussion by eyaler** [google/or-tools#3283] [Feature requests] [answered]
**What language and solver does this apply to?**
Python / CP-SAT

**Describe the problem you are trying to solve.**
I am minimizing an objective, then adding the best value as a constraint and solving to get all solutions with that value.
For this I need to clear the objective by either creating a new solver or calling model.Proto().ClearField('objective')

**Describe the solution you'd like**
I suggest adding an explicit ClearObjective() / RemoveObjective() to the API thus circumventing the need to access Proto() as well as being more discoverable and concise. Also, I would suggest adding a remark to the documentation regarding the necessity of this step in the above use case.

**Comment by lperron** [comment, marked as answer]
Implemented on master.

Thanks
